### PR TITLE
ETCM-8753 use deregistration in wizard

### PR DESCRIPTION
**Pull request by kpinter-iohk**
# Description

Add your description here, if it fixes a particular issue please provide a
[link](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword=)
to the issue.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages.
- [ ] New tests are added if needed and existing tests are updated.
- [ ] Relevant logging and metrics added
- [ ] CI passes. See note on CI.
- [ ] Any changes are noted in the `changelog.md` for affected crate
- [ ] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG Partner Chains developers to do this
for you.

### CODE REVIEW

File: toolkit/partner-chains-cli/src/deregister/mod.rs [hunk 1]
```diff
@@ -1,100 +1,57 @@
...
 	fn run<C: IOContext>(&self, context: &C) -> anyhow::Result<()> {
-		let (chain_config, cardano_network) = read_chain_config_file(context)?;
+		let chain_config = read_chain_config_file(context)?;
 		context.print(
 		  &format!("This wizard will remove the specified candidate from the committee candidates based on the following chain parameters:\n{}.\nCommittee Candidate Validator Address is '{}'\n",
           &chain_config.chain_parameters,
           &chain_config.cardano_addresses.committee_candidates_address)
     	);
-		let (payment_signing_key_path, cold_vkey) = get_cardano_keys(context)?;
+		context.print("Payment signing key and verification key of cold key used for registration are required to deregister.");
```
Comment (LGLO) [inline toolkit/partner-chains-cli/src/deregister/mod.rs:26]: Cosmetic: "verification key of cold key" => "cold verification key"

File: toolkit/partner-chains-cli/src/deregister/mod.rs [hunk 2]
```diff
@@ -1,100 +1,57 @@
...
+		let runtime = tokio::runtime::Runtime::new().map_err(|e| anyhow::anyhow!(e))?;
+		runtime
+			.block_on(offchain.deregister(
+				chain_config.chain_parameters.genesis_utxo,
+				payment_signing_key,
+				stake_ownership_pub_key,
+			))
+			.map_err(|e| anyhow::anyhow!("Candidate registration failed: {e:?}!"))?;
+
+		Ok(())
```
Comment (LGLO) [inline toolkit/partner-chains-cli/src/deregister/mod.rs:45]: Cosmetic: is it required? Isn't removing `?;` two lines above enough?

File: toolkit/partner-chains-cli/src/deregister/tests.rs [hunk 3]
```diff
@@ -86,20 +103,16 @@ fn fails_when_cold_key_is_not_valid() {
...
 }
 
-/// Read the chain configuration file twice, because for some reason cardano network is not present in ChainConfig.
-fn read_config_twice_io() -> MockIO {
-	MockIO::Group(vec![
-		MockIO::file_read(CHAIN_CONFIG_FILE_PATH),
-		MockIO::file_read(CHAIN_CONFIG_FILE_PATH),
-	])
+fn read_config_once_io() -> MockIO {
+	MockIO::Group(vec![MockIO::file_read(CHAIN_CONFIG_FILE_PATH)])
```
Comment (LGLO) [inline toolkit/partner-chains-cli/src/deregister/tests.rs:115]: This is very convoluted way of saying this thing.

Either leave just `MockIO::file_read(CHAIN_CONFIG_FILE_PATH)` or inline it call sites, please.

**Review by LGLO — APPROVED** [review]
Few cosmetic issues only

File: toolkit/partner-chains-cli/src/deregister/mod.rs (hunk 2 above)
Comment (kpinter-iohk) [reply toolkit/partner-chains-cli/src/deregister/mod.rs:45]: Unfortunately it is, as `deregister` returns `Option<McTxHash>` and `run` has to return `Result<()>`

**Pull request merged**